packages/@okta/vuepress-site/docs/reference/api/roles/index.md

```diff
@@ -14,11 +14,1064 @@ The Okta Administrator Roles API provides operations to manage administrative ro
 
 Explore the Administrator Roles API:  [![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/4f1233beeef282acbcfb)
 
+## Custom role operations
+<ApiLifecycle access="beta" />
+These operations allow the creation and manipulation of custom roles as custom collections of [permissions](#permission-types).
+
+### Create role
+<ApiOperation method="post" url="/api/v1/iam/roles" />
+
+Creates a new role with a custom set of permissions
+
+#### Request parameters
+
+| Parameter   | Description                          | Param Type   | DataType                    | Required |
+| :---------- | :----------------------------------- | :----------- | :----------------------------------------------- | :------- |
+| label       | name given to new role               | Body         | String                                           | TRUE     |
+| description | description of the new role          | Body         | String                                           | TRUE     |
+| permissions | the permissions the new role grants  | Body         | Array of [Permission types](#permission-types)   | TRUE     |
+
+#### Response parameters
+
+Created custom [role](#custom-role-object)
+
+#### Request example
+
+```bash
+curl -v -X POST \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+-d '{
+      "label": "UserCreator",
+      "description": "...",
+      "permissions" [
+        "okta.users.create",
+        "okta.users.read",
+        "okta.groups.read",
+        "okta.users.profile.manage"
+      ]
+    }' "https://${yourOktaDomain}/api/v1/iam/roles"
+```
+
+#### Response example
+
+```json
+{
+  "id": "cr0Yq6IJxGIr0ouum0g3",
+  "label": "UserCreator",
+  "description": "Create users",
+  "permissions": [
+    "okta.users.create",
+    "okta.users.read",
+    "okta.groups.read",
+    "okta.users.profile.manage"
+  ]
+}
+```
+
```

Comment:
looking at this and our existing docs, we don't seem to callout error responses very much - is it necessary/desirable to do so - i.e. "if permissions are not provided ..." or "if a custom role with this name already exists" ... etc?

Reply:
as discussed offline, we can call out specific errors if needed but otherwise, as long as the errors are typical errors expected from a REST endpoint, we won't usually doc those.